feg/gateway/services/n7_n40_proxy/servicers/create_session_test.go

```diff
@@ -0,0 +1,336 @@
+	require.NoError(t, err)
+	return &n7.N7Config{
+		DisableN7: false,
+		Server: n7.PCFConfig{
+			ApiRoot:      *apiRoot,
+			TokenUrl:     TOKEN_URL,
+			ClientId:     CLIENT_ID,
+			ClientSecret: CLIENT_SECRET,
+		},
+		Client: n7.N7ClientConfig{
+			LocalAddr:     LOCAL_ADDR,
+			NotifyApiRoot: NOTIFY_API_ROOT,
+		},
+	}
+}
+
+func newCreateSessionRequest(imsi string) *protos.CreateSessionRequest {
+	return &protos.CreateSessionRequest{
+		CommonContext: &protos.CommonSessionContext{
+			Sid:     &protos.SubscriberID{Id: imsi},
+			RatType: protos.RATType_TGPP_NR,
+			UeIpv4:  UE_IPV4,
+			Apn:     APN1,
+		},
+		RatSpecificContext: &protos.RatSpecificContext{
+			Context: &protos.RatSpecificContext_M5GsmSessionContext{
+				M5GsmSessionContext: &protos.M5GSMSessionContext{
+					PduSessionId:   uint32(PDU_SESSION_ID),
+					Gpsi:           GPSI1,
+					PduSessionType: protos.PduSessionType_IPV4,
+				},
+			},
+		},
+		SessionId:      genSessionID(IMSI1),
+		AccessTimezone: &protos.Timezone{OffsetMinutes: 330},
+	}
+}
+
+func genSessionID(imsi string) string {
+	return fmt.Sprintf("%s-1234", imsi)
+}
+
+func matchSmPolicyContextData(imsi string) interface{} {
+	return func(reqBody n7_sbi.PostSmPoliciesJSONRequestBody) bool {
+		return reqBody.Supi == sbi.Supi(imsi) &&
+			reqBody.NotificationUri == n7.GenNotifyUrl(NOTIFY_API_ROOT, genSessionID("IMSI"+imsi))
+	}
+}
+
+func newSmPolicyResponse(t *testing.T) *n7_sbi.PostSmPoliciesResponse {
+	policyDecisionStr := `{
+		"pccRules": {
+			"rule1": {
+				"pccRuleId": "rule1",
+				"flowInfos": [{
+					"flowDescription": "permit in ip from 0.0.0.0/0 to 4.2.2.4"
+				}],
+				"precedence": 1,
+				"refQosData": ["qos_data1"],
+				"refTcData": ["tc_data1"],
+				"refChgData": ["chg_data1"],
+				"refUmData": ["um_data1"],
+				"refCondData": "cond_data1"
+			},
+			"static_rule1": {
+				"pccRuleId": "static_rule1",
+				"refCondData": "cond_data1"
+			}
+		},
+		"qosDesc": {
+			"qos_data1": {
+				"qodId": "qos_data1",
+				"5qi": 5,
+				"maxbrUl": "200000",
+				"maxbrDl": "500000",
+				"gbrUl": "100000",
+				"gbrDl": "250000"
+			}
+		},
+		"chgDecs": {
+			"chg_data1": {
+				"chgId": "chg_data1",
+				"online": true,
+				"ratingGroup": 1,
+				"serviceId": 12
+			}
+		},
+		"traffContDecs": {
+			"tc_data1": {
+				"tcId": "tc_data1",
+				"redirectInfo": {
+					"redirectEnabled": true,
+					"redirectAddressType": "URL",
+					"redirectServerAddress": "https://redirect.example.com/tc"
+				}
+			}
+		},
+		"umDecs": {
+			"um_data1": {
+				"umId": "um_data1",
+				"volumeThreshold": 4000000,
+				"volumeThresholdUplink": 1500000,
+				"volumeThresholdDownlink": 3500000
+			}
+		},
+		"conds": {
+			"cond_data1": {
+				"condId": "cond_data1",
+				"activationTime": "2021-10-22T12:42:31Z",
+				"deactivationTime": "2021-10-22T14:42:31Z"
+			}
+		},
+		"policyCtrlReqTriggers": ["RE_TIMEOUT"],
+		"revalidationTime": "2021-10-22T14:42:31Z",
+		"online": true
+	}`
```

Comment:
see you can use placeholders in case you need to change some parameters

for example 

```
policyDecisionStr : = "pccRules": {
			"rule1": {
				"pccRuleId": "%s",
				....
				}

policyDecisionStr = fmt.Sprintf(policyDecisionStr, "ppc_rule_1")
```
You don't need to add in each field, but just in case you need to reuse `policyDecisionStr` in other places where the response may be different

Reply:
Added some placeholders